dali/ft/filecopy.cpp

```diff
@@ -3224,8 +3224,70 @@ void FileSprayer::updateTargetProperties()
             }
 
             // Keep source kind
-            curProps.setProp("@kind", srcAttr->queryProp("@kind"));
+            if (srcAttr->hasProp("@kind"))
+                curProps.setProp("@kind", srcAttr->queryProp("@kind"));
+            else
+            {
+                const char * targetKind = "unknown";
+
+                if (tgtFormat.markup == FMTxml)
+                    targetKind = "xml";
+                else if (tgtFormat.markup == FMTjson)
+                    targetKind = "json";
+
+                const char * targetFormat = "unknown";
+
+                switch (tgtFormat.type)
+                {
+                case FFTfixed:
+                case FFTvariable:
+                case FFTblocked:
+                    targetKind = "flat";
+                    break;
+                case FFTcsv:
+                    targetKind = "csv";
+                    break;
+                case FFTutf:
+                    targetFormat = "utf8n";
+                    break;
+                case FFTutf8:
+                    targetFormat = "utf8";
+                    break;
+                case FFTutf16:
+                    targetFormat = "utf16";
+                    break;
+                case FFTutf16be:
+                    targetFormat = "utf16be";
+                    break;
+                case FFTutf16le:
+                    targetFormat = "utf16le";
+                    break;
+                case FFTutf32:
+                    targetFormat = "utf32";
+                    break;
+                case FFTutf32be:
+                    targetFormat = "utf32be";
+                    break;
+                case FFTutf32le:
+                    targetFormat = "utf32le";
+                    break;
+                case FFTrecfmvb:
+                    targetFormat = "recfmvb";
+                    break;
+                case FFTrecfmv:
+                    targetFormat = "recfmv";
+                    break;
+                case FFTvariablebigendian:
+                    targetFormat = "variablebigendian";
+                    break;
+                }
 
+                if (! streq(targetKind, "unknown"))
+                    curProps.setProp(FPkind, targetKind);
```

Comment:
better to set targetKind to nullptr and avoid the streq to "unknown", i.e. make them:
```
if (targetKind)
    curProps.setProp(FPkind, targetKind);
```

Reply:
Changed